server/matcher/match.go

```diff
@@ -37,6 +38,9 @@ func New() *Matcher {
 // lot size, unless the order is a market buy order, which is not subject to
 // this constraint.
 func orderLotSizeOK(ord order.Order, lotSize uint64) bool {
+	// if ordRev, isOR := ord.(*OrderRevealed); isOR {
+	// 	ord = ordRev.Order
+	// }
```

Comment:
This is an example of why `OrderRevealed` does not embed `order.Order`.  There were many opportunities for bugs with functions that accept an `order.Order`, but don't know what to do with a `matcher.OrderRevealed.